varats-core/varats/provider/patch/patch_provider.py

```
@@ -157,6 +180,46 @@ def __str__(self) -> str:
 
         return str_representation
 
+    def render(self, **kwargs) -> Path:
+        """
+        Renders the patch with the given arguments.
+
+        Args:
+            render_args: Arguments to render the patch with
+
+        Returns:
+            Path to the rendered patch
+        """
+        if not self.arguments:
+            return self.path
+
+        render_args = self.arguments.copy()
+        for key, value in kwargs.items():
+            render_args[key] = value
```

Should we fail if `kwargs` contains arguments that are not specified by the patch?

---

Hmm, hard to say. We could fail or just emit a warning.

Could it be that one collects a larger set of arguments and puts it into different patches and hence sometimes puts args in that are not "needed"?

---

I can see the following cases why there might be arguments not specified by the patch:

1. The case Flo described (If you want to apply/render multiple patches that share some arguments and don't overlap in the others it might be easier to just aggregate all arguments into one dict as opposed to building them per-patch)
2. Someone misspelled an argument name
3. An argument in the template was not specified in the patch info.

For 1. I don´ t see a big issue allowing these extra arguments. There might  be other, more general pitfalls when someone does this (e.g. accidentally overriding arguments for other patches) but that cannot really be solved in the rendering step.

For 2. the rendering would fail later regardless, so having a warning and a subsequent exception raised in the rendering is sufficient.

Case 3. is where I see potential issues in the future, when people just forget/ignore to properly add arguments to the patch info and work around it in their local setup. 

I am unsure whether the benefit of allowing it for 1. outweighs the potential problems caused by 3. But for 3. we could also setup other measures to avoid this (e.g. proper CI in the patches repository)